### PR TITLE
raftstore: speedup clear meta

### PR DESCRIPTION
**Pull request by BusyJay**
The keys to be deleted are known when clearing meta, hence create two iterators are unnecessary.

### CODE REVIEW

**Comment by javaforfun** [comment]
LGTM

File: src/raftstore/store/store.rs [hunk 1]
```diff
@@ -294,13 +294,18 @@ impl<T, C> Store<T, C> {
...
-        }
+        let raft_state: RaftLocalState =
+            match self.engine.get_msg_cf(CF_RAFT, &raft_key).unwrap() {
+                // it has been cleaned up.
+                None => return,
+                Some(value) => value,
+            };
+        let apply_key = keys::apply_state_key(region.get_id());
+        let apply_state: RaftApplyState =
+            self.engine.get_msg_cf(CF_RAFT, &apply_key).unwrap().unwrap();
```
Comment (javaforfun) [inline src/raftstore/store/store.rs]: If restart happen when CompactLog applied (apply_state changed), but RaftlogGcTask not run.  We will get wrong first_index and dirty data after clear_meta?

File: src/raftstore/store/peer_storage.rs [hunk 2]
```diff
@@ -988,37 +998,28 @@ impl PeerStorage {
...
+        try!(wb.delete_cf(handle, &keys::raft_log_key(region_id, id)));
+    }
+    try!(wb.delete_cf(handle, &keys::raft_state_key(region_id)));
+    try!(wb.delete_cf(handle, &keys::apply_state_key(region_id)));
+ 
     info!("[region {}] clear peer {} meta keys and {} raft keys, takes {:?}",
           region_id,
-          meta_count,
-          raft_count,
+          1,
```
Comment (siddontang) [inline src/raftstore/store/peer_storage.rs]: 1 meta keys?

**Comment by BusyJay** [comment]
PTAL

**Comment by javaforfun** [comment]
LGTM

File: src/raftstore/store/peer_storage.rs (hunk 2 above)
Comment (siddontang) [inline src/raftstore/store/peer_storage.rs]: ping

Comment (siddontang) [inline src/raftstore/store/peer_storage.rs]: I prefer distinguishing raft keys and raft logs in the log.

File: src/raftstore/store/peer_storage.rs [hunk 3]
```diff
@@ -988,37 +994,36 @@ impl PeerStorage {
...
-                        &raft_start,
-                        &raft_end,
+                        &begin_log_key,
+                        &end_log_key,
                         false,
                         &mut |key, _| {
-                            try!(wb.delete_cf(handle, key));
-                            raft_count += 1;
-                            Ok(true)
+                            first_index = keys::raft_log_index(key).unwrap();
```
Comment (siddontang) [inline src/raftstore/store/peer_storage.rs]: why not use the first index from the raft state directly?

Comment (BusyJay) [inline src/raftstore/store/peer_storage.rs]: ref: https://github.com/pingcap/tikv/pull/2095#discussion_r130013690

**Comment by siddontang** [comment]
maybe we can use delete range directly.

Now, LGTM

**Pull request merged**